servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -308,16 +305,22 @@ public Reference assignReference(
       ReferenceType referenceType, String referenceName, String expectedHash, Reference assignTo)
       throws NessieNotFoundException, NessieConflictException {
     try {
-      checkArgument(expectedHash != null, "Expected hash must be provided");
 
       ResolvedHash oldRef =
           getHashResolver()
               .resolveHashOnRef(
-                  referenceName, expectedHash, "Expected hash", REQUIRED_UNAMBIGUOUS_HASH);
+                  referenceName,
+                  expectedHash,
+                  new HashValidator("Reference to assign from", "Expected hash")
```

Comment:
maybe just `Reference to assign`?.. not sure I get the meaning of `from` in the message :thinking:

Reply:
`Assignment target` SGTM